### PR TITLE
feat: ramps for multiple environments

### LINKED ISSUE #25664: [Feature]: Ramps for ME in CE codebase
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Summary

Job story for displaying ramps in CE for BE only feature ME.

### Why should this be worked on?

For release of ME

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
PR to add business ramps to CE for multiple environments. These will be behind a feature flag.

Adds ramps for following areas
- When you create/edit datasource, you'll see a ramp for the users in the form of env switching sidebar
- In every page, you'll see a switch in the bottom bar and the staging config will always be disabled. User will not be able to switch to the staging configuration. 

#### PR fixes following issue(s)
Fixes #25664 

#### Type of change

- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?

- [x] Manual
- [x] Jest
- [ ] Cypress

#### Test Plan

- _**Tried with user who can see ME**_

**Switch env dropdown**
1. Ramps seen on canvas/preview/deploy mode
2. Tooltip link takes to upgrade url on all three modes
3. Staging option cannot be accessed from dropdown - just a ramp is visible
4. Compared with Figma on this switch and raised a couple of UI issues [tooltip position on switch environment dropdown
size difference in texts] 

**Side bar**
1. Ramps seen on Side bar

- _**User not exposed to ME**_

1. Cannot see The env switch bar itself. 
2. User cannot see the sidebar in Edit DS page.

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by ChandanBalajiBP — APPROVED** [review]

**Comment by btsgh** [comment]
Issues images - 
**Tooltip position**

dp
zeplin

**switch environment dropdown size difference in texts**
Screenshot 2023-07-31 at 6 14 40 PM

**Pranay's observation -** 
https://ru2fwekpma.vmaker.com/record/IEgbqZnOc0BtZ3d5

**Comment by btsgh** [comment]
@ayushpahwa - I see this behavior when I share the app with Developer and Admin rights. This seems different from what is mentioned in the matrix that was shared by you. Let me know your thoughts. 

https://github.com/appsmithorg/appsmith/assets/101863839/ed99ca25-0671-46c5-8e97-cf5873d2955c

**Review by ChandanBalajiBP — APPROVED** [review]
LGTM, but please take approval for RAMPS for concerned dev.

**Review by berzerkeer — APPROVED** [review]

**Comment by btsgh** [comment]
As per update from @ayushpahwa - This is waiting on approval from Parth for the ramps matrix functionality.

**Comment by btsgh** [comment]
As per the slack [conversation](https://theappsmith.slack.com/archives/C0423TJFUJK/p1691119994211969?thread_ts=1691067958.876799&cid=C0423TJFUJK), current implementation will be continued, and the issue found will be addressed separately.

File: app/client/src/ce/constants/messages.ts [hunk 1]
```diff
@@ -1874,7 +1874,9 @@ export const MOVE_TO_BUSINESS_EDITION = (trailingChar: string) =>
 export const START_SWITCH_ENVIRONMENT = (environment: string) =>
   `Switching your environment to ${environment}, and running all associated pageload actions`;
 export const SWITCH_ENVIRONMENT_SUCCESS = (environment: string) =>
-  `Environment switched to ${environment}`;
+  `...Environment switched to ${environment} successfully`;
```
Comment (ChandanBalajiBP) [inline app/client/src/ce/constants/messages.ts:1877]: can you please remove `...`

Comment (ayushpahwa) [reply app/client/src/ce/constants/messages.ts]: Fixed, please check

**Review by ChandanBalajiBP — APPROVED** [review]

**Pull request merged**